### ISSUE TITLE
Option to disable the list continuation feature

### ISSUE THREAD

**Issue by MagTun** [yzhang-gh/vscode-markdown#439]
#### What is the problem?
+ When I write "1)" or "-" at the beginning of a line and then press `enter` the "1)" or the "-" is deleted and the cursor goes to the next line
+ I have use some ahk shortcuts and they are not process correctly:
for instance, when I write "ddd", ahk transforms this into 2019_05_08, but you extension transforms this into `.md` file this becomes "dd2019_05"

This bugs only happens on a `.md` file and if your extension is activated. If I uninstall your extension and install another Markdown linting, I don't have these bugs. 

![2019-05-08_13-59-28](https://user-images.githubusercontent.com/15843700/57373825-a42f2980-7199-11e9-9f47-bdb665c58ca7.gif)

#### How can I reproduce it?
Cf above 

#### Is there any error message in the console?
No 

Extension version: 2.3.1
VS Code version: Code 1.33.1 (51b0b28134d51361cf996d2f0a1c698247aeabd8, 2019-04-11T08:27:14.102Z)
OS version: Windows_NT x64 10.0.17763

**Comment by yzhang-gh** [comment]
For the first problem, it is by design (removing an empty list item). May I know why you need `1)` or `-` on a single line?

The second one is hard. It has been reported in #200.
The Vim extension also has this problem. Up to now, we don't have a solution (even don't know why).

**Comment by MagTun** [comment]
I just wanted to prepare a template of several list items. I would have filled it later (ex: in a meeting) . That would be great if there was a custom setting for this. 
Sorry for the duplicate, I didn't find it.

**Comment by yzhang-gh** [comment]
> I just wanted to prepare a template of several list items. I would have filled it later (ex: in a meeting)

An interesting use case.

However, I would say it is not that necessary (to have a template), as the Markdown syntax is already simple enough.

Just type <code>- </code>&nbsp;and start writing. You will get an auto-added <code>- </code>&nbsp;before the next list item.

![enter](https://github.com/yzhang-gh/vscode-markdown/raw/master/images/gifs/on-enter-key1.gif)

When you finish, just press <kbd>Enter</kbd> again. The empty <code>- </code>&nbsp;will be removed.

![empty list](https://github.com/yzhang-gh/vscode-markdown/raw/master/images/gifs/on-enter-key2.gif)

There are even more advantages, like list reordering

![reorder](https://github.com/yzhang-gh/vscode-markdown/raw/master/images/gifs/marker-fixing.gif)

Don't bother to prepare a template. Just write Markdown on the fly 😉.

**Comment by MagTun** [comment]
Thanks for the little tuto because I wasn't aware of this!

But still it would be nice to have the option to disable. For instance if I want someone to write 3 choices for a questions, I can just write 
```
1)
2)
3) 
```

instead of having to write 

> (3 answers please)

**Comment by yzhang-gh** [comment]
Makes sense. Before we really have this option, there is a workaround you can have a try.

Open keybinding settings (find it in the left bottom corner), search for <kbd>Ctrl</kbd>+<kbd>Enter</kbd> and right-click the extension contributed one. Then you can insert a new line with <kbd>Ctrl</kbd>+<kbd>Enter</kbd>.

![image](https://user-images.githubusercontent.com/7588612/57550308-e99c5400-735d-11e9-88d2-118416378836.png)

**Comment by MagTun** [comment]
That's just perfect, that way I can have it both way! This is even better than what I wanted!
It's just too bad, you have this problem with the "text snippet" with AHK because you extension is really great!

**Comment by t1anchen** [comment]
> 
> Just type `- ` and start writing. You will get an auto-added `- ` before the next list item.
> 

A common scenario for me is the combination use of multiple line cursors and markdown mode. Suppose there is a list with multiple line cursors located at the end of each line, when `enter` or `ctrl+enter` is pressed, the actual action appears to merely insert new list item at first cursor. If the expected action is to insert new list item for each line (or simply inserting new line is acceptable), is it feasible to implement the feature by configuring the markdown mode or PRs required?

**Comment by yzhang-gh** [comment]
Currently, we don't support multi-cursors in this list continuation feature. Quite a few changes are needed to implement this.
